runtime/src/main/java/io/quarkiverse/rabbitmqclient/RabbitMQReadyCheck.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkiverse.rabbitmqclient;
+
+import java.net.Socket;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.Readiness;
+
+import com.rabbitmq.client.Address;
+
+/**
+ * RabbitMQ ready check which checks if at least one of the configured brokers is available.
+ *
+ * @author b.passon
+ */
+@Readiness
+@ApplicationScoped
+public class RabbitMQReadyCheck implements HealthCheck {
+
+    @Inject
+    RabbitMQClientConfig config;
+
+    @Override
+    public HealthCheckResponse call() {
+        if (atLeastOneBrokerIsAlive()) {
+            return HealthCheckResponse.up("At least one RabbitMQ broker is available.");
+        }
+        return HealthCheckResponse.down("No RabbitMQ broker is available.");
```

Comment:
You should check that all broker are alive and set the readiness check to DOWN if one is not alive.
This is what is usuallly done for exemple for database when multiple datasource.
Also,  you should add some data() with some info on the status of the check.

Reply:
Added data and checking all brokers now.